### DISCUSSION TITLE
NemaMessage does not include empty GPGGA message

### DISCUSSION THREAD

**Discussion by lluatehwachs** [dotMorten/NmeaParser#105] [Q&A] [answered]
Hi, 

I am using SharpGIS.NmeaParser Nuget 2.2.2 in a Windows 10 dot net core WPF project.  All other messages are shown correctly except the empty GPGGA message.  Seems that empty GPGGA messages are filtered out. Is this known?  

Some sample messages from termnal:
$GPGGA,,,,,,,,,,M,,M,,*56
$PTNL,GGK,133435.00,071502,0000.00000000,N,00000.00000000,E,0,00,0.0,EHT0.000,M*7E
$GPVTG,,T,,M,,N,,K,N*2C
$GPGST,,,,,,,,*57
$GPGSV,,,*79
$GPZDA,133435.01,15,07,2002,00,00*67
$GPGSA,A,1,,,,,,,,,,,,,0.0,0.0,0.0*30
$GPRMC,133435.01,V,,,,,,,150702,,,N*7E
$GPGLL,,,,,,,*7C
$GPGNS,,,,,,,,,,,,*4D
$GPGGA,,,,,,,,,,M,,M,,*56
$PTNL,GGK,133436.00,071502,0000.00000000,N,00000.00000000,E,0,00,0.0,EHT0.000,M*7D

Thanks,
Liang

**Comment by dotMorten** [comment, marked as answer]
The gga message looks incorrect. There should be a * before the last two characters. Only messages that are valid and with a correct checksum are raised

**Reply by lluatehwachs** [reply]
Thank you so much!

**Comment by lluatehwachs** [comment]
Hi, 
Seems that empty GGA message  "$GPGGA,,,,,,,,,,M,,M,,*56" was not showing either. 

import pynmea2
>>> msg=pynmea2.parse("$GPGGA,,,,,,,,,,M,,M,,*56")
>>> msg
<GGA(timestamp=None, lat='', lat_dir='', lon='', lon_dir='', gps_qual=None, num_sats='', horizontal_dil='', altitude=None, altitude_units='M', geo_sep='', geo_sep_units='M', age_gps_data='', ref_station_id='')>

**Comment by dotMorten** [comment]
Fixed in https://github.com/dotMorten/NmeaParser/commit/91d0a73cd29cb1a262532fef25baf1f45d1b64ce
